lti_consumer/lti.py

```diff
@@ -128,8 +128,18 @@ def get_signed_lti_parameters(self):
             u'lis_result_sourcedid': self.xblock.lis_result_sourcedid,
 
             u'context_id': self.xblock.context_id,
+            u'custom_component_display_name': self.xblock.display_name,
         }
 
+        if self.xblock.due is not None:
+            lti_parameters.update({
+                u'custom_component_due_date': self.xblock.due.strftime('%Y-%m-%d %H:%M:%S')
+            })
+            if self.xblock.graceperiod is not None:
+                lti_parameters.update({
+                    u'custom_component_graceperiod': self.xblock.graceperiod.total_seconds(),
+                })
```

Comment:
Sorry, couple more things here I missed. You can just set the value on the `lti_parameters` dict here instead of calling `update`. You can leave out the `is not None` in the if statements because `None` is falsy.
```
if self.xblock.due:
        lti_parameters['custom_component_due_date'] = self.xblock.due.strftime('%Y-%m-%d %H:%M:%S')
        if self.xblock.graceperiod:
            lti_parameters['custom_component_graceperiod'] = self.xblock.graceperiod.total_seconds()
```